keep-ui/app/(keep)/topology/ui/map/service-node.tsx

```diff
@@ -73,34 +72,52 @@ function ServiceDetailsTooltip({ data }: { data: TopologyService }) {
 }
 
 export function ServiceNode({ data, selected }: NodeProps<ServiceNodeType>) {
-  const { useAllAlerts } = useAlerts();
-  const { data: alerts, mutate } = useAllAlerts("feed");
-  const { data: pollAlerts } = useAlertPolling();
+  const { data: incidents, mutate: incidentsMutate } = useIncidents(
+    true,
+    25,
+    0,
+    {
+      id: "creation_time",
+      desc: false,
+    },
+    { affected_services: [data.display_name] }
+  );
```

Comment:
This sends out a request to get incidents for *every* service in the topology map. Sounds a bit problematic in terms of performance

Reply:
fixing this.

Reply:
Fixed.